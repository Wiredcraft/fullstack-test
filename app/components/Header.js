import React, { PropTypes } from 'react';
import styles from '../less/header.less';

export default function Header(props) {
    return (
        <div className={styles.header}>
            <div className={styles.logo} onClick={props.onBackToIndex}>
                {props.logoText}
            </div>
            <div className={styles.status} onClick={props.onRedirect}>
                {props.status}
            </div>
        </div>
    );
}

Header.propTypes = {
    logoText: PropTypes.string,
    status: PropTypes.string,
    onBackToIndex: PropTypes.func,
    onRedirect: PropTypes.func
};