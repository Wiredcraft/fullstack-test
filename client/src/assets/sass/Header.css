@import url("https://fonts.googleapis.com/css?family=Roboto+Condensed:300,300i,400");
.c-Header {
  background-color: #FF5D73;
  padding: 10px;
  display: flex;
  line-height: 1.5;
  justify-content: space-between;
  color: #ffffff; }
  .c-Header__title {
    margin: 0;
    font-size: 1.3rem; }
  .c-Header__add-btn {
    background-color: transparent;
    font-size: 1rem;
    font-family: "Roboto Condensed", sans-serif; }
    .c-Header__add-btn:focus {
      outline: 0;
      background-color: #ff2a47; }
