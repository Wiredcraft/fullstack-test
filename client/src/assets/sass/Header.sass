@import './_variables.sass';

.c-Header
  background-color: $primary-color
  padding: 10px
  display: flex
  line-height: 1.5
  justify-content: space-between
  &__title
    margin: 0
    font-size: 1.3rem
  &__add-btn
    background-color: transparent
    font-size: 1rem
    border: 1px solid $text-color
    font-family: "Roboto Condensed", sans-serif
    &:focus
      outline: 0
      background-color: darken($primary-color, 10%)
