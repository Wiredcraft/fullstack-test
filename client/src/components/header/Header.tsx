import * as React from "react";

import { useAuth } from "../../contexts/AuthContext";
function Header() {
	const { isAuthenticated } = useAuth() // to decide whether is going to show login button or create talk button
	const [showLoginForm, setLoginState] = React.useState(false)
	const [showCreateTalkForm, setTalkState] = React.useState(false)
	
	return (
		<div>
			{
				if(!isAuthenticated){
					return <button >Open Login Form By Changing state</button>
				}
			}

			{
				if(isAuthenticated){
					return <button >Open Create talk form by Changing state</button>
				}
			}

			{
				if(showCreateTalkForm) {
					return <LoginForm />
				}
				if(showCreateTalkForm) {
					return <CreateTalkForm />
				}
			}

		</div>
	)
}