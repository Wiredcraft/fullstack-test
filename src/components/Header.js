import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { withRouter } from 'react-router';

class Header extends Component {
  render() {
    return (
      <div>
        <h2>Hello from header</h2>
      </div>
    );
  }
}

export default withRouter(Header);
