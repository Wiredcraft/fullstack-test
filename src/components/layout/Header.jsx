import React from 'react';
import { Link } from 'react-router-dom';
import { useNavigate } from 'react-router-dom';


const Header = () => {
    const navigate = useNavigate()
    function goToHome(){
        window.location.href= window.location.origin
    }
    function logout(){
        sessionStorage.removeItem("user")
        sessionStorage.removeItem("token")
        sessionStorage.removeItem("userId")
        goToHome()
    }

    return (
        <header style={headerStyle}>
            <div style={divStyle}>
                <h1 style={{cursor:"pointer"}} onClick={goToHome}>Lightning Talk</h1>
                {
                    sessionStorage.getItem("user")?
                    <div style={divStyle}>
                      <span>{`Welcome ${sessionStorage.getItem("user")}!`}</span>
                      <button style={btnStyle} onClick={logout}>Logout</button>
                    </div>:
                    <Link style={linkStyle} to="/user">Sign Up/Login</Link>

                }
            </div>
        </header>
    )
};

const btnStyle = {
    color: '#4867AA',
    background: "white",
    display: 'inline-block',
    border: 'none',
    padding: '7px 20px',
    cursor: 'pointer'
};

const divStyle = {
    display: "flex",
    alignItems: "flex-end",
    justifyContent: "space-between"
};

const headerStyle = {
    background: "#4867AA",
    color: 'white',
    textAlign: 'center',
    padding: '10px'
};

const linkStyle = {
    color: 'white',
    textDecoration: 'none'
};

export default Header;