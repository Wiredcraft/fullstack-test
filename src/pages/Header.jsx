import React, { Component } from 'react'
import { Redirect } from 'react-router-dom'
import LightingTalkItem from '../components/LightingTalkItem'
import '../assets/css/index.css'

class Home extends Component {
  constructor () {
    super()
    this.state = {
      data: null
    }
    this.voteHandler = this.voteHandler.bind(this)
  }

  componentDidMount() {
    // fetch('/api/lightingTalks', { method: 'GET' })
    //   .then(res => res.json())
    //   .then(data => {
    //     this.setState({ data })
    //   })
    //   .catch(err => {
    //     console.log(err)
    //   })
  }

  voteHandler (id) {
    // const formData = new FormData()
    // formData.append('id', id)
    // console.log(formData)
    // fetch('/api/vote', { method: 'PUT', body: {"id": id}})
    //   .then(res => res.json())
    //   .then(data => {
    //     this.setState({ data })
    //   })
    //   .catch(err => {
    //     console.log(err)
    //   })
  }

  render() {
    return (
      <div>
        <h1 className='title'>lighting talks</h1>
      </div>
    )
  }
}

export default Home