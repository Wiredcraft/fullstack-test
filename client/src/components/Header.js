import React from 'react';
import '../assets/sass/Header.css'

const Header = (props) =>
  <div className='c-Header'> 
    <h1 className='c-Header__title'>Hacker Talks</h1>
    <button
      className='c-btn'
      onClick={props.onAddClick}
    >
      &#10133; add
    </button>
  </div>

export default Header;
