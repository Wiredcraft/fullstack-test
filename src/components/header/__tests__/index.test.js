import React from 'react';
import renderer from 'react-test-renderer';

import Header from '../index';

jest.mock('react-router-dom');

test('Header renders', () => {
  const component = renderer.create(<Header />);

  expect(component.toJSON()).toMatchSnapshot();
});
