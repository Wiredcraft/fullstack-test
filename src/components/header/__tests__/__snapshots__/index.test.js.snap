// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders 1`] = `
<div
  className="sc-bdvvtL kygHzC"
>
  <span
    className="sc-gsDKAQ ka-Dnaq"
    onClick={[Function]}
  >
    ⚡️ Lightning Talks
  </span>
  <button
    className="sc-dkPtRN bIiLjj"
    onClick={[Function]}
  >
    Add talk
  </button>
</div>
`;
